### PR TITLE
useSelect: silently error (for block zombie children)

### LINKED ISSUE #29498: Deleting first item in an InnerBlocks list with delete key causes exception
## Description

If the delete key or keyboard shortcut is used to delete the first item in an InnerBlocks list an exception is thrown. Using the 'Remove block' menu option for the block works as expected. Using the delete key for any block other than the first one also works as expected.

This issue can be replicated with the Navigation block, and also on the [gallery block refactor PR](https://github.com/WordPress/gutenberg/pull/25940/)

## Step-by-step reproduction instructions:
1. Add a Navigation block and add several child navigation items to it.
2. Select the first item in the navigation and hit delete key to remove
3. Block shows unable to preview message and exception in console

deletekey

## Code snippet 

Exception is thrown at https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/use-block-props/index.js#L86 

For some reason the useBlockProps hook for the block that has been deleted gets stuck in a loop and throws an exception when trying to get the title of the block that is now deleted.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #29498. The problem is that `useSelect` child subscriptions are called first, so the parent block list still thinks the block is present. 

Quick solution: `useSelect` should silently error (so the crash boundary is not triggered) and revert to the latest map output. There's no need to retry mapping since the block is deleted and the subscription will be removed at a later point.

I'll add an e2e test.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/data/src/components/use-select/index.js [hunk 1]
```diff
@@ -148,12 +148,11 @@ export default function useSelect( _mapSelect, deps ) {
...
-
-				throw new Error( errorMessage );
-			} else {
-				// eslint-disable-next-line no-console
-				console.error( errorMessage );
 			}
+
+			// eslint-disable-next-line no-console
+			console.error( errorMessage );
+			mapOutput = latestMapOutput.current;
```
Comment (ellatrix) [inline packages/data/src/components/use-select/index.js:155]: I'll add a comment here in a bit.

**Comment by youknowriad** [comment]
> There's no need to retry mapping since the block is deleted

I wonder if this is the only potential trigger for zombie bugs. Maybe sometimes it's a prop that is outdated compared to what's in store and need to wait for the parent to update the prop first.

File: packages/data/src/components/use-select/index.js [hunk 2]
```diff
@@ -148,12 +148,11 @@ export default function useSelect( _mapSelect, deps ) {
 				errorMessage += `\nThe error may be correlated with this previous error:\n`;
 				errorMessage += `${ latestMapOutputError.current.stack }\n\n`;
 				errorMessage += 'Original stack trace:';
-
-				throw new Error( errorMessage );
```
Comment (youknowriad) [inline packages/data/src/components/use-select/index.js:152]: Does this mean we're basically silencing all errors from useSelect?

**Comment by ellatrix** [comment]
@youknowriad Highly likely. This is just a quick fix to prevent a block crash. :)

File: packages/data/src/components/use-select/index.js (hunk 2 above)
Comment (ellatrix) [reply packages/data/src/components/use-select/index.js:152]: Errors will be logged, not thrown no.

**Review by youknowriad — APPROVED** [review]
I'm still hesitant about the silenced errors but let's do it and see.

**Comment by ellatrix** [comment]
Who originally wanted to do this retry thing? @nerrad? I'm not sure if you meant that if this once worked and now broke, or if it was an idea, or if this bug just isn't covered by retrying.

**Comment by youknowriad** [comment]
> Who originally wanted to do this retry thing?

It was originally suggested and implemented by @epiqueras taking inspiration from react-redux. I believe it worked at that time but maybe didn't survive the different use-select refactors.

**Pull request merged**